src/server/routes/system/health.ts

```diff
@@ -52,6 +53,7 @@ export async function healthCheck(fastify: FastifyInstance) {
       res.status(StatusCodes.OK).send({
         status: "OK",
         engineVersion: process.env.ENGINE_VERSION,
+        engineTier: process.env.ENGINE_TIER,
```

Comment:
Once we update TIER for all cloud-hosted, can we assume the Engine is self-hosted if this field is undefined?

If so, maybe this can fallback to `?? 'SELF_HOSTED'`